### PR TITLE
v1.8.2 release

### PR DESCRIPTION
**Pull request by juanrhenals**
Enhancement - Added a functional test for a now fixed TaskEngine deadlock issue.
Enhancement - Extend the TaskEngine Interface by adding the GetTaskByArn method.
Enhancement - Improved general stability for integ tests.
Enhancement - Disabled OOM testing for unsupported kernel configurations.
Bug - Fixed an issue where exec_create and exec_start events were not correctly ignored with some Docker versions.
Bug - Resolved a bug where sending a signal to a container caused the agent to treat the container as dead.

r? @aaithal  @samuelkarp

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## 1.8.2
+* Enhancement - Added a functional test for a now fixed TaskEngine deadlock issue.
```
Comment (samuelkarp) [inline CHANGELOG.md]: I think several of these are not changelog-worthy as they are not user-facing.  In general, things that affect only developers of the agent are reasonable to exclude from the changelog.

I would remove the following:
* Enhancement - Added a functional test for a now fixed TaskEngine deadlock issue.
* Enhancement - Extend the TaskEngine Interface by adding the GetTaskByArn method.
* Enhancement - Improved general stability for integ tests.
* Enhancement - Disabled OOM testing for unsupported kernel configurations.

I would change the duplicate reporting one to read as follows:
* Bug - Fixed an issue where `exec_create` and `exec_start` events were not correctly ignored with some Docker versions.

I would also change the `kill` one to read as follows:
* Bug - Resolved a bug where sending a signal to a container caused the agent to treat the container as dead.

Comment (juanrhenals) [inline CHANGELOG.md]: Sounds good. I'll update the changelog to:

Bug - Fixed an issue where exec_create and exec_start events were not correctly ignored with some Docker versions.
Bug - Resolved a bug where sending a signal to a container caused the agent to treat the container as dead.

**Comment by juanrhenals** [comment]
Updated CHANGELOG.md

**Comment by samuelkarp** [comment]
:+1:

**Comment by samuelkarp** [comment]
Can you rebase on top of `dev`?

**Comment by juanrhenals** [comment]
Rebased and updated changelog.

**Comment by samuelkarp** [comment]
:+1:

**Comment by aaithal** [comment]
lgtm

**Pull request merged**